flake.lock

```diff
@@ -0,0 +1,98 @@
+{
```

Comment:
I'm not really a fan of these unrelated files landing in the root of the repo.
I'd prefer to have these non-rust changes split out from this PR (as they don't relate to the main thesis of this PR), and either not have them in the repo at all, or to be encapsulated in a sub-directory if possible.  For example, do they make sense to put under a directory like `assets/nix`?

Reply:
i'm pretty sure that would work, i just need them to exist under version control otherwise nix flakes cannot properly manage my developer environment

Reply:
@AusCyberman for the flakes, when it's untracked you can tell git you intend to add it **but without staging the files** right away, with `git add -N` (or the long form `git add --intend-to-add`)
It'swhat I do for a few projects where I want to have a Nix flake but not commit it for a PR

Reply:
i've already added them tho, so what now?

Reply:
To clean up this PR from Nix stuff, I'd recommend something like this:
```sh
git rm --cached flake.*  # remove files from git
git commit -m "Remove Nix flakes files" flake.*
git add -N flake.*  # re-add them to git without staging them
```

Reply:
Hmm and you may have a few more files to remove, like cargo.nix and maybe also crate-hashes.json ? (I'm not entirely familiar with the rust toolchains in Nix)